### ISSUE TITLE
Discrepancies in time logging

### ISSUE THREAD

**Issue by jan-imrich** [jkraemer/stopwatch#6]
We are using the redmine stopwatch plugin for managing time spent on various projects by various developers (lets say five people using stopwatch at one time on various redmine issues).
Unfortunately, we are experiencing some issues. The time recorded by stopwatch often does not correspond with the actual time the person spent working on the subject. Most of the time, when the issue appears, there is approximately half an hour missing, sometimes more. This is happening few times a day.
I've already added logging to the update function, printing all of the variables available. Unfortunately, I haven't caught the issue in log output yet - there is a lot of traffic and none of the developers reported demonstrable discrepancy since.
We have theory, that there could be some fault in calculation of the runtime hours when the number becomes larger (more than 1 hour?) - for example when the developer is working and does not have the redmine issue or modal window open, and the update function is not called until he decides to finally open the redmine issue again.
Could it possibly be an race condition in the update function? The function may be called quite often, possibly working with old values, before the previous request updates them (start_time)? This would probably result in more hours logged (logging the same amount of time twice), not in a loss though.

I could add screenshots of redmine issue, where the issue is apparent (I started the stopwatch on certain time, let it alone, and then stopped the stopwatch approximately three hours later. There is an hour missing). But I see no point in doing this, since I have no log output from this otherwise certain discrepancy.

```
Environment:
  Redmine version                4.1.1.stable
  Ruby version                   2.6.6-p146 (2020-03-31) [x86_64-linux]
  Rails version                  5.2.4.2
  Environment                    production
  Database adapter               Mysql2
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
SCM:
  Subversion                     1.10.4
  Mercurial                      4.8.2
  Bazaar                         2.8.0
  Git                            2.20.1
  Filesystem                     
  GitMirror                      2.20.1
Redmine plugins:
  additionals                    3.0.0
  periodictask                   4.1.0
  redmine_autoclose              0.0.4
  redmine_checklists             3.1.18
  redmine_dmsf                   2.4.4
  redmine_git_mirror             0.8.0
  redmine_pivot_table            0.0.7
  stopwatch                      0.1.0
  vault                          0.4.1


```
What could I do to debug this issue? 
Thanks for developing such an useful redmine plugin. I hope I could help to make it even better.
Jan.

**Comment by jan-imrich** [comment]
I think I've found the culprit. 
In the update function, the :hours column is updated based on sum of the entry.hours getter and time spent since started_at (previous update). Problem is, the .hours getter does rounding to two decimal places. Since the update function is possibly called quite often (every minute, or so), the rounding could result in significant logged time discrepancy at the end.
See how I resolved it in https://github.com/jkraemer/stopwatch/pull/7
